.github/workflows/inspektor-gadget.yml

```diff
@@ -227,6 +227,27 @@ jobs:
         registry: ${{ env.REGISTRY }}
         container-image: ${{ env.CONTAINER_REPO }}
         co-re: ${{ matrix.type == 'core' }}
+    - name: Get btfhub-archive last commmit
+      id: get-btfhub-head
+      run: |
+        echo "head=$(git ls-remote https://github.com/aquasecurity/btfhub-archive/ HEAD | cut -f1)" >> $GITHUB_OUTPUT
+      shell: bash
+    - name: Cache BTFGen
+      id: cache-btfgen
+      uses: actions/cache@v3.0.11
+      with:
+        path: hack/btfs
+        # key is composed by
+        # - arch: amd64 or arm64
+        # - lastest commit of btfhub
+        # - hash of all bpf objects
+        key: ${{ matrix.platform }}-${{ steps.get-btfhub-head.outputs.head }}-${{ hashFiles('pkg/gadgets/*/*/tracer/*.o') }}
```

Comment:
I found couple of locations that might need to handled as well [`piditer`](https://github.com/inspektor-gadget/inspektor-gadget/tree/main/pkg/gadgets/top/ebpf/piditer) and [`container-map`](https://github.com/inspektor-gadget/inspektor-gadget/tree/main/pkg/gadgettracermanager/containers-map). (I wonder if we have can a check to ensure above pattern does catches all the object files. Probably can be handled in separate PR)

Reply:
nit: Prefix the key with `bpfgen-` it is easier to validate it in [caches](https://github.com/inspektor-gadget/inspektor-gadget/actions/caches).

Reply:
I changed the pattern to match all object files within pkg/gadgets. The container-map doesn't need this information, so it's not included. I also added the btfgen prefix to the cache key.

Reply:
It seems `piditer` files are still missing:

```
→ la -la pkg/gadgets/*/*/tracer/*.o  | wc -l
40
→ la -la pkg/gadgets/*/*/*/*.o  | wc -l
42

```

Should we change the pattern to ` pkg/gadgets/*/*/*/*.o ` ?

Reply:
It turns out I pushed the wrong branch. I tested with the `mauricio/backups/cachebtfgen` branch on https://github.com/mauriciovasquezbernal/inspektor-gadget/pull/4/files but this PR uses the `mauricio/cache-btfgen` branch. I pushed the right code for this branch.